src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -129,7 +129,7 @@ private CompilationWithAnalyzers(Compilation compilation, ImmutableArray<Diagnos
             _cancellationToken = cancellationToken;
 
             _compilationData = new CompilationData(_compilation);
-            _analysisState = new AnalysisState(analyzers, _compilationData);
+            _analysisState = new AnalysisState(analyzers, _compilationData, _compilation.Options);
```

Comment:
but you could expose? you already added option so might be okay but adding one more parameter seems more work than just exposing option from compilation data.